src/main/java/seedu/address/model/entitylist/ParticipantList.java

```diff
@@ -50,6 +50,15 @@ public Participant get(Id id) throws AlfredException {
      * @return boolean
      */
     public boolean update(Id id, Participant updatedParticipant) {
+        // Also check if new Participant with updated details exists already
+        // i.e. update John to Joshua, but list already contains Joshua
+        // AB3 had a isSamePerson() method
+        // so maybe we can have a isSameParticipant() in the Participant class (and rest of the entities)
+        /*
+         * for each p in this.participants
+         *     if p.isSameParticipant(updatedParticipant)
+         *         return false
+         */
```

Comment:
This cross checking is done in my PR i think

Reply:
Is it? Could you tell me where?

Reply:
Okay not exactly the same use case, but it does it adding new entities into the list. For the rest, we have overrriden the equals class.